pcdsdevices/slits.py

```diff
@@ -0,0 +1,321 @@
+        """
+        return (self.xwidth.position, self.ywidth.position)
+
+    def remove(self, size=None, wait=False, timeout=None, **kwargs):
+        """
+        Open the slits to unblock the beam
+
+        Parameters
+        ----------
+        size : float, optional
+            Open the slits to a specific size otherwise
+            `:attr:`.nominal_aperture is used
+
+        wait : bool, optional
+            Wait for the status object to complete the move before returning
+
+        timeout : float, optional
+            Maximum time to wait for the motion. If None, the default timeout
+            for this positioner is used
+
+        Returns
+        -------
+        MoveStatus:
+            Status object based on move completion
+
+        See Also
+        --------
+        :meth:`Slits.move`
+        """
+        # Use nominal_aperture by default
+        size = size or self.nominal_aperture
+        return self.move(size, wait=wait, timeout=timeout, **kwargs)
+
+    def set(self, size):
+        """
+        Alias for the move method, here for bluesky compatibilty
+        """
+        return self.move(size, wait=False)
+
+    @property
+    def hints(self):
+        """Device hints"""
+        return {'fields': [self.xwidth.readback.name,
+                           self.ywidth.readback.name]}
+
+    def open(self):
+        """
+        Uses the built-in `OPEN` record to move open the aperture
+        """
+        self.open_cmd.put(1)
+
+    def close(self):
+        """
+        Close the slits to have an aperture of 0mm on each side
+        """
+        self.close_cmd.put(1)
+
+    def block(self):
+        """
+        Overlap the slits to block the beam
+        """
+        self.block_cmd.put(1)
+
+    def stage(self):
+        """
+        Store the initial values of the aperture position before scanning
+        """
+        self._original_vals[self.xwidth.setpoint] = self.xwidth.readback.value
+        self._original_vals[self.ywidth.setpoint] = self.ywidth.readback.value
+        return super().stage()
+
+    def subscribe(self, cb, event_type=None, run=True):
+        """
+        Subscribe to changes of the slits
+
+        Parameters
+        ----------
+        cb : callable
+            Callback to be run
+
+        event_type : str, optional
+            Type of event to run callback on
+
+        run : bool, optional
+            Run the callback immediatelly
+        """
+        # Avoid making child subscriptions unless a client cares
+        if not self._has_subscribed:
+            # Subscribe to changes in aperture
+            self.xwidth.readback.subscribe(self._aperture_changed,
+                                           run=False)
+            self.ywidth.readback.subscribe(self._aperture_changed,
+                                           run=False)
+            self._has_subscribed = True
+        super().subscribe(cb, event_type=event_type, run=run)
```

Comment:
This should return the callback id from `super().subscribe`

Reply:
Will do